### PR TITLE
:seedling:  Update destroy function 

### PR DESCRIPTION
**Pull request by Sajiyah-Salat**
## Description
Updated destroy function to run docker only when image name is present. This removes the log error while generating docs.

Partial address: https://github.com/kubernetes-sigs/kubebuilder/issues/3435

### CODE REVIEW

File: test/e2e/utils/test_context.go [hunk 1]
```diff
@@ -232,9 +232,12 @@ func (t *TestContext) Tidy() error {
...
 	//nolint:gosec
-	cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
-	if _, err := t.Run(cmd); err != nil {
-		warnError(err)
+	// if image name is not present or not provided skip execution of docker command
+	if t.ImageName != "" {
+		cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
+		if _, err := t.Run(cmd); err != nil {
+			warnError(err)
+		}
```
Comment (varshaprasad96) [inline test/e2e/utils/test_context.go:240]: Looks good! Seems like when `t.ImageName` is set by hardcoding the value in some of the places while creating a test context, like this: https://github.com/kubernetes-sigs/kubebuilder/blob/a5b246f1ff5b6444ef75bb82c73ae557f47b2438/test/e2e/utils/test_context.go#L91.

So let's also add a warning when `t.ImageName` is empty and we don't execute the command. Something like this:

```suggestion
	if t.ImageName != "" {
		cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
		if _, err := t.Run(cmd); err != nil {
			warnError(err)
		}
	} else {
	           warnErr(fmt.Errorf("ImageName not set, skipping cleaning up of docker image"))
	}	
```

Wdyt? 
cc: @camilamacedo86 ?

Comment (Sajiyah-Salat) [reply test/e2e/utils/test_context.go:240]: if there is no image we are already giving a warning error then why adding else.

File: test/e2e/utils/test_context.go [hunk 2]
```diff
@@ -232,9 +232,12 @@ func (t *TestContext) Tidy() error {
 // Destroy is for cleaning up the docker images for testing
 func (t *TestContext) Destroy() {
 	//nolint:gosec
-	cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
-	if _, err := t.Run(cmd); err != nil {
-		warnError(err)
+	// if image name is not present or not provided skip execution of docker command
+	if t.ImageName != "" {
```
Comment (camilamacedo86) [inline test/e2e/utils/test_context.go:236]: We could either use trim here to avoid the edge case " " string.Trim(t.ImageName) and then you check the len to see if is < 0. 

However, it is a nit I am fine to move forward with as it is .

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

/approved

Since followed up the @varshaprasad96 suggestions.

File: test/e2e/utils/test_context.go [hunk 3]
```diff
@@ -232,9 +232,12 @@ func (t *TestContext) Tidy() error {
...
+		cmd := exec.Command("docker", "rmi", "-f", t.ImageName)
+		if _, err := t.Run(cmd); err != nil {
+			warnError(err)
```
[the remark this answers is not in the document]
Comment (Sajiyah-Salat) [reply test/e2e/utils/test_context.go:239]: > @varshaprasad96 Isnt this the line where we are warning about error. I am not sure that I understood the codeblock but it looks like if any error occur we are giving the warning but its not specified for not passing the image name.

I did this comment immediately after your reply. But there was a pending label and I thought you havent read it yet. It was really confusing and now I got it.
